dspace-server-webapp/src/test/java/org/dspace/app/rest/csv/CSVMetadataImportReferenceTest.java

```diff
@@ -0,0 +1,328 @@
+     */
+    @Test
+    public void testMultiRowNameRef() throws Exception {
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,dc.identifier.other,rowName",
+                "+,Person,," +  col1.getHandle() + ",0,val1",
+                "+,Person,," +  col1.getHandle() + ",1,val2",
+                "+,Publication,rowName:val1||rowName:val2," +  col1.getHandle() + ",2,val3"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[2], items[0], 1);
+        assertRelationship(items[2], items[1], 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * a UUID reference
+     */
+    @Test
+    public void testSingleUUIDReference() throws Exception {
+        Item person = ItemBuilder.createItem(context, col1)
+                .withRelationshipType("Person")
+                .build();
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,rowName,dc.identifier.other",
+                "+,Publication," + person.getID().toString() + "," +  col1.getHandle() + ",anything,0"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[0], person, 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * a UUID reference with multiable references
+     */
+    @Test
+    public void testMultiUUIDReference() throws Exception {
+        Item person = ItemBuilder.createItem(context, col1)
+                .withRelationshipType("Person")
+                .build();
+        Item person2 = ItemBuilder.createItem(context, col1)
+                .withRelationshipType("Person")
+                .build();
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,rowName,dc.identifier.other",
+                "+,Publication," + person.getID().toString() + "||" + person2.getID().toString() + "," +
+                        col1.getHandle() + ",anything,0"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[0], person, 1);
+        assertRelationship(items[0], person2, 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * multi metadata references. One archived item and one in the csv.
+     */
+    @Test
+    public void testMultiRefArchivedCsv() throws Exception {
+        Item person = ItemBuilder.createItem(context, col1)
+                .withTitle("Person")
+                .withRelationshipType("Person")
+                .build();
+        String[] csv = {"id,dc.title,relationship.type,relation.isAuthorOfPublication,collection,rowName," +
+                "dc.identifier.other",
+                "+,Person2,Person,," +  col1.getHandle() + ",idVal,0",
+                "+,Pub1,Publication,dc.title:Person||dc.title:Person2," +  col1.getHandle() + ",anything,1"};
```

Comment:
Does this also support mixing a metadata, UUID, rowName lookup, and still retain the order?

Reply:
Yes it should, and we will add a test case for mixed ref types